src/infql.jl

```diff
@@ -176,8 +171,7 @@ function materialize!(M::Lmul{<:AdjQLPackedQLayout{<:BandedColumns},<:PaddedLayo
 end
 
 function _lmul_cache(A::Union{AbstractMatrix{T},AbstractQ{T}}, x::AbstractVector{S}) where {T,S}
-    TS = promote_op(matprod, T, S)
-    lmul!(A, cache(convert(AbstractVector{TS},x)))
+    lmul!(A, cache(x))
```

Comment:
Why was this change made? It means it won't be correct if the type of `Q*x` is different than `x`

Reply:
`promote_op` caused some type inference issues but I think they're resolved now so I have added it back